stdlib/test/utils/test_stringref.mojo

```diff
@@ -108,26 +108,6 @@ def test_find():
     assert_equal(StringRef("").find("abc"), -1)
 
 
-def test_endswith():
-    var empty = StringRef("")
-    assert_true(empty.endswith(""))
-    assert_false(empty.endswith("a"))
-    assert_false(empty.endswith("ab"))
-
-    var a = StringRef("a")
-    assert_true(a.endswith(""))
-    assert_true(a.endswith("a"))
-    assert_false(a.endswith("ab"))
-
-    var ab = StringRef("ab")
-    assert_true(ab.endswith(""))
-    assert_false(ab.endswith("a"))
-    assert_true(ab.endswith("b"))
-    assert_true(ab.endswith("b", start=1))
-    assert_true(ab.endswith("a", end=1))
-    assert_true(ab.endswith("ab"))
```

Comment:
**Request** Could we move these tests to `test_string_slice.mojo`? From a quick look, it seems like we don't currently have dedicated tests for `StringSlice.endswith()`. Should be a quick copy/paste + replace `StringRef` → `StringSlice` 🙂

Reply:
Sorry I didn't look for them, I though whoever moved the implementations took the tests with them

Reply:
No worries, totally understand, it's an easy oversight to make 🙂  I only double checked because my spidey tense started tingling 🕷️, I had optimistically assumed they'd have been copied over too.